static/js/components/ExpandedCourseDisplay.js

```diff
@@ -0,0 +1,140 @@
+// @flow
+/* global SETTINGS: false */
+import React from "react"
+import _ from "lodash"
+import moment from "moment"
+import striptags from "striptags"
+import { AllHtmlEntities } from "html-entities"
+
+import type { Course } from "../flow/discussionTypes"
+import { embedlyResizeImage } from "../lib/url"
+import { maxPrice } from "../lib/courses"
+import { platforms } from "../lib/constants"
+import { languageName } from "../lib/util"
+import ClampLines from "react-clamp-lines"
+
+const COURSE_IMAGE_DISPLAY_HEIGHT = 300
+const entities = new AllHtmlEntities()
+
+type Props = {
+  course: Course
+}
+
+export default class ExpandedCourseDisplay extends React.Component<Props> {
```

Comment:
this component could be written as a stateless (functional) component instead of a class. it might be good to split up the JSX into a few helper components (they could still live in this file) just to make it easier to read, but that's up to you.

Reply:
I'd prefer to leave it as a class, more consistent with `ExpandedPostDisplay`

Reply:
ok, it's mostly a style thing either way, but React is able to perform some optimizations with stateless components that are not possible with class-based components - the performance difference is probably not big, but I tend to prefer them for that reason where possible.